web/src/message_util.ts

```diff
@@ -16,23 +19,33 @@ export function do_unread_count_updates(messages, expect_no_new_unreads = false)
     }
 }
 
-export function add_messages(messages, msg_list, opts) {
+export function add_messages(
+    messages: RawMessage[],
+    msg_list: MessageList,
+    append_to_view_opts: {messages_are_new: boolean},
+): MessageListAddMessagesReturn {
     if (!messages) {
         return undefined;
     }
 
     loading.destroy_indicator($("#page_loading_indicator"));
 
-    const render_info = msg_list.add_messages(messages, opts);
+    const render_info = msg_list.add_messages(messages, append_to_view_opts);
 
     return render_info;
 }
 
-export function add_old_messages(messages, msg_list) {
+export function add_old_messages(
+    messages: RawMessage[],
+    msg_list: MessageList,
+): MessageListAddMessagesReturn {
     return add_messages(messages, msg_list, {messages_are_new: false});
 }
 
-export function add_new_messages(messages, msg_list) {
+export function add_new_messages(
+    messages: RawMessage[],
```

Comment:
I had trouble figuring out where to use `RawMessage` and where to use `Message`, and it seems like I'll have to convert between them at some point? `message_list_data` has a bunch of things that use `Message` that this file interacts with, and `update_expected_max_message_id` expects `RawMessage`. 

@andersk do you know how to deal with this off the top of your head? Would you be down to pair on it?

Reply:
from [here](https://github.com/zulip/zulip/pull/28414/commits/52ce46d7f5a5738c0e2370f7da8dc4ef1aba371b#r1440878961) Tim said:

> I don't think these are `RawMessage` objects;
> 
> At least, they've all been processed via `message_helper.process_new_message` ... which probably would be a good one to convert to `TypeScript` and typed as converting `RawMessage` into `Message` objects? I've not double-checked but it seems like that might be what it does.

Reply:
I tried switching from `RawMessage` to `Message` in `fetch_status.ts` and that keeps the types consistent for now. I think that might actually be correct, in that `process_message` is actually processing them before calls to `fetch_status` happen (and updating the object directly) but it's not super clear in `message_helper` right now that this is actually happening.

Reply:
`message_helper` is actually pretty soon in the queue of files I can work on, so I can go back to it then

Reply:
Yeah I think code paths like `message_events.insert_new_message` need to rename their input variable as `raw_message`, which it immediately converts into a `Message` object, in order to type-check.